test/NuGet.Core.FuncTests/NuGet.Signing.CrossFramework.Test/CrossFrameworkVerificationTest.cs

```diff
@@ -0,0 +1,408 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Signing.CrossFramework.Test
+{
+    [Collection(CrossVerifyTestCollection.Name)]
+    public class CrossFrameworkVerificationTest
```

Comment:
Excluding the tests that only sign in netfx and verify on netcoreapp, the list of tests here are: 
```text
Verify_AuthorSignedPackage_SuccessAsync
Verify_AuthorSignedTimestampedPackage_SuccessAsync
Verify_RepositorySignedPackage_SuccessAsync
Verify_RepositorySignedTimestampedPackage_SuccessAsync
Verify_AuthorSignedRepositoryCounterSignedPackage_SuccessAsync
Verify_AuthorSignedTimestampedRepositoryCounterSignedPackage_SuccessAsync
Verify_AuthorSignedRepositoryCounterSignedTimestampedPackage_SuccessAsync
Verify_AuthorSignedTimestampedRepositoryCounterSignedTimestampedPackage_SuccessAsync
```

Basically this is a matrix of:
* author signed, repo signed, both
* with and without timestamping on each

These tests run in a new CI job in parallel to much slower tests, so it's not really a big problem here, but I feel like this duplicate a lot of testing. For example, I can't imagine a scenario where author signed with timestamp and repo signed with timestamp fails, but all the other tests pass.

We should write tests as if the thing we're testing is a black box (don't write tests that only someone who read the implementation could know is an edge case), but I think we should also be mindful about whether the test can actually catch a regression that no other tests can. Our CI job already runs for a very long time, so I'd like us to avoid writing tests that won't help us catch bugs.

Reply:
Thanks for the suggestion!
I think we are testing the combination as they are using different properties:
The primary signature is a SignedCms.SignerInfos Property of the .signature.p7s (signedCms).
The timestamp on the primary signature is a UnsignedAttributes property of the SignerInfo of primary signature.
The counter signature is a CounterSignerInfos property of the primary signature.
The timestamp on the counter signature is a UnsignedAttributes property of the SignerInfo of counter signature.
So it might be safer to check if all the properties could be read correctly cross framework.